nix/modules/scripts.nix

```diff
@@ -35,16 +35,34 @@
         set -xeuo pipefail
         trap "cd $PWD" EXIT
 
-        cd versions/0_1
-        nix flake update
-        cd ../../
+        export VERSIONS_DIR="./versions/''${1}"
+        export DEFAULT_VERSIONS_DIR="$(nix flake metadata --no-write-lock-file --json | jq --raw-output '.locks.nodes.versions.locked.path')"
 
-        nix flake lock --update-input versions --override-input versions "path:./versions/0_1"
+        (
+          cd "$VERSIONS_DIR"
+          nix flake update --tarball-ttl 0
+        )
 
-        if [[ $(${pkgs.git}/bin/git diff -- flake.lock versions/*/flake.lock | grep -E '^[+-]\s+"' | grep -v lastModified --count) -eq 0 ]]; then
-          echo got no actual source changes, reverting modifications..;
-          ${pkgs.git}/bin/git checkout flake.lock versions/*/flake.lock
+        if [[ $(${pkgs.git}/bin/git diff -- "$VERSIONS_DIR"/flake.lock | grep -E '^[+-]\s+"' | grep -v lastModified --count) -eq 0 ]]; then
+          echo got no actual source changes, reverting modifications..
+          ${pkgs.git}/bin/git checkout $VERSIONS_DIR/flake.lock
+          exit 0
+        else
+          git add "$VERSIONS_DIR"/flake.lock
         fi
+
+        if [[ "$VERSIONS_DIR" = "$DEFAULT_VERSIONS_DIR" ]]; then
+          nix flake lock --tarball-ttl 0 --update-input versions --override-input versions "path:$VERSIONS_DIR" 
+        fi
+
+        if [[ $(${pkgs.git}/bin/git diff -- flake.lock | grep -E '^[+-]\s+"' | grep -v lastModified --count) -eq 0 ]]; then
```

Comment:
Is there a difference between `${pkgs.git}/bin/git` and `git`?

Reply:
yes, just `git` is impure and relies on `git` being available in `PATH`. `${pkgs.git}/bin/git` evaluates to an absolute path to the binary.

Reply:
that said, i haven't been consistent in using one or the other.